pkg/api/rpc.go

```diff
@@ -58,3 +64,14 @@ type CollectResponse struct {
 }
 
 type HealthcheckResponse = HealthcheckReport
+
+type TaskStatusResponse struct {
+	Priority   int
+	ID         string
+	Type       string
+	Input      interface{}
+	Result     task.TaskResult
+	Created    string
+	LastUpdate string
+	LastState  string
+}
```

Comment:
I would prefer to directly send `task.Task` but there's an issue when mapstructure decodes `time.Time` which can be [easily fixed](https://github.com/mitchellh/mapstructure/issues/159). But there's another, more complicated issue, to decode enums (for `TaskType` and `TaskState`) which I don't know how to solve yet. We could use reflection, but reflecting the enum type will only give me `string` which is not very helpful.

Reply:
Yeah, that struct seems weird. Why not just define a JSON Marshal/Unmarshal methods, rather than store time and state as strings? Also maybe rename `LastUpdate` to `Updated`, and `LastState` to `State`.